### PR TITLE
feat: assimilate the gcm client code

### PR DESCRIPTION
**Pull request by jrconlin**
Closes: #1057

### CODE REVIEW

File: autopush/router/gcm.py [hunk 1]
```diff
@@ -115,19 +115,24 @@ def _route(self, notification, uaid_data):
         try:
             gcm = self.gcm[router_data['creds']['senderID']]
             result = gcm.send(payload)
+        except RouterException:
```
Comment (pjenvey) [inline autopush/router/gcm.py]: these 2 lines are redundant

File: autopush/router/gcmclient.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        response = self._sender(
+            url=self._endpoint,
+            headers=headers,
+            data=json.dumps(payload.payload),
+            **self._options
+        )
+
+        if response.status_code == 400:
```
Comment (pjenvey) [inline autopush/router/gcmclient.py]: nitpick: could collapse the next section with this one `response.status_code in (400, 404)`

File: autopush/tests/test_router.py [hunk 3]
```diff
@@ -296,9 +296,177 @@ def check_results(result):
         return d
 
 
+class GCMClientTestCase(unittest.TestCase):
```
Comment (pjenvey) [inline autopush/tests/test_router.py]: Could we put this TestCase in a new file? test_router's already pretty hefty

File: autopush/router/gcm.py (hunk 1 above)
Comment (jrconlin) [reply autopush/router/gcm.py]: No, they're not. There's a general Exception handler further down that transposes unexpected exceptions into RouterExceptions with logging.

File: autopush/tests/test_router.py (hunk 3 above)
Comment (jrconlin) [reply autopush/tests/test_router.py]: 👍

**Pull request merged**